ooni/nettest.py

```diff
@@ -329,7 +331,7 @@ def setupTestCases(self, test_cases):
         self.testClasses = set([])
         self.testHelpers = {}
 
-        if config.reports.unique_id is True:
+        if config.reports.unique_id is True and not self.reportID:
```

Comment:
This was actually a bug that lead to there being a mismatch between the report_id being generated server-side and the one that was submitted by the client.

Basically we want to support having a report_id also when we write to disk, but we don't want the report_id to be ovewritten by the client if the backend sets it.

Overall if you look through the diffs you will see that the code pertaining to handling the reportID is very fragile and convoluted and would need some better refactoring, but for the moment it works ok.